source/task5/task5.html.md.erb

```diff
@@ -0,0 +1,185 @@
+
+2. Make the prototype directory into a local Git repo:
+
+    ```shell
+    git init
+    ```
+
+### Add and commit your prototype files
+
+1. Add all the files in the local repo and stage them for commit:
+
+    ```shell
+    git add .
+    ```
+
+    You can check to see what files have been added by running:
+
+    ```shell
+    git status
+    ```
+    
+2. Commit the staged files:
+
+    ```shell
+    git commit -m "COMMIT-MESSAGE"
+    ```
+
+    `COMMIT-MESSAGE` is the message describing the commit. This is your first commit to the repo so you can replace `COMMIT-MESSAGE` with `first-commit`.
+
+
+### Push the staged commits to the remote repo
+
+1. Go to the remote repo in GitHub. This is the empty repo you created earlier.
+
+2. Select the __Code__ button.
+
+3. Select the button next to the url to copy the url to your clipboard.
+
+4. Back in the terminal, link the local repo to the remote repo:
+
+    ```shell
+    git remote add origin REMOTE-REPO-URL
+    ```
+
+    `REMOTE-REPO-URL` is the url you copied.
+    
+5. Push the changes in your local repo to the remote repo:
+
+    ```shell
+    git push -u origin master
+    ```
+
+You have now created a remote version of your prototype on GitHub. Go to the remote repo and check that all your files are there.
+
+> Accounts with two-factor authentication (2FA) will not be able to push commits first time and will need to generate a personal access token.
+
+## GitHub and 2FA
+
+If your GitHub account has 2 Factor Authentication (2FA) turned on, you might need to create a personal access token to push your first commit to a remote repo.
+
+In the terminal, you'll be asked for your:
+
+- username, which will be the same one you used to log into your GitHub account
+- password, which will be a personal access token created from within your GitHub account
+
+1. Select your profile photo on any GitHub page, then select __Settings__.
+
+2. Select __Developer settings__.
+
+3. Select __Personal access tokens__.
+
+4. Select __Generate new token__.
+
+    You may be asked for your account password at this point.
+
+5. Give your token a name and select the __repo__ checkbox.
+
+    This gives you full access to private and public repositories from within the command line.
+
+6. Select the __Generate token__ button once you're done.
+
+7. To use the token, select the __copy__ icon to copy the token to your clipboard.
+
+    Save this somewhere safe. Once you navigate away from the page you will no longer be able to view it.
+
+You can now use the token in the terminal to perform git operations:
+
+```shell
+git clone https://github.com/username/repo.git
+Username: your_username
+Password: your_token
+```
+
+## Deploy with Heroku
+
+1. Go to [www.heroku.com](Heroku) and log in if you're not already logged in.
+
+2. Select **New**, then **Create new app**.
```

Comment:
We seem to mix using `__` and `**` for emphasis in this file – is that worth fixing whilst we're tidying things up?

Reply:
Good plan, thanks!